### PR TITLE
server/ui: associate user timing profile

### LINKED ISSUE #51: Add a way to associate user accounts to a timing profile
We have `TimingAuthor` because we have karaokes made by people that are not users of karaberus (inactive/did not migrate, imported karaokes, ...). But we should associate Users to a TimingAuthor (already the case in the DB model) so we can add the timing author of the current user when uploading the subtitles, list issues assigned to our karaokes, ...

We'll probably still need some way to not set the author when doing minor changes to sub files.

### PR DESCRIPTION
**Pull request by NextFire**
Fix #51 

Preferred merge strategy: squash

### CODE REVIEW

File: server/karaberus.go [hunk 1]
```diff
@@ -90,6 +90,9 @@ func addRoutes(api huma.API) {
 	huma.Get(api, "/api/token", GetAllUserTokens, setSecurity(oidc_security))
 	huma.Post(api, "/api/token", CreateToken, setSecurity(oidc_security))
 	huma.Delete(api, "/api/token/{token}", DeleteToken, setSecurity(oidc_security))
+
+	huma.Get(api, "/api/me", GetMe, setSecurity(oidc_security))
```
Comment (odrling) [inline server/karaberus.go:94]: maybe add a route to get other users

File: server/user.go [hunk 2]
```diff
@@ -37,3 +37,35 @@ func getOrCreateUser(ctx context.Context, sub string, info *oidc.UserInfo) (*Use
 	}
 	return &user, nil
 }
+
+type GetMeOutput struct {
+	Body User
```
Comment (odrling) [inline server/user.go:42]: might be the first time we expose it in the API, we can set the fields of the struct to have nicer names when serialized in json.

File: server/karaberus.go [hunk 3]
```diff
@@ -90,6 +90,9 @@ func addRoutes(api huma.API) {
...
+	huma.Put(api, "/api/me/author", UpdateMeAuthor, setSecurity(oidc_security))
```
Comment (odrling) [inline server/karaberus.go:95]: It actually can edit other users and “me” is a default which is odd, `/api/users/{id}/author` sounds better to me.

File: server/user.go [hunk 4]
```diff
@@ -37,3 +37,35 @@ func getOrCreateUser(ctx context.Context, sub string, info *oidc.UserInfo) (*Use
...
+}
+
+func GetMe(ctx context.Context, input *struct{}) (*GetMeOutput, error) {
+	user := getCurrentUser(ctx)
+	return &GetMeOutput{Body: *user}, nil
+}
+
+type UpdateMeAuthorInput struct {
+	Body struct {
+		Id *uint `json:"id"`
```
Comment (odrling) [inline server/user.go:52]: already mentionned above but probably should be a path parameter

**Review by odrling — CHANGES_REQUESTED** [review]

File: server/user.go [hunk 5]
```diff
@@ -38,13 +38,40 @@ func getOrCreateUser(ctx context.Context, sub string, info *oidc.UserInfo) (*Use
...
+}
+
+func GetMe(ctx context.Context, input *struct{}) (*GetUserOutput, error) {
+	return getUser(ctx, nil)
+}
+
+func getUser(ctx context.Context, sub *string) (*GetUserOutput, error) {
+	out := &GetUserOutput{}
+	if sub != nil {
+		user, err := getOrCreateUser(ctx, *sub, nil)
```
Comment (odrling) [inline server/user.go:60]: get or create?

Comment (NextFire) [reply server/user.go]: It no longer upserts the user when info is nil

**Review by odrling — APPROVED** [review]

**Pull request merged**